lib/src/iterable.dart

```diff
@@ -458,10 +458,29 @@ extension IterableX<E> on Iterable<E> {
     return list.sublist(length - n);
   }
 
+  /// Returns a list containing the elements at every [n]th index, starting
+  /// from the index [offset].
+  ///
+  /// ```dart
+  /// var chars = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+  /// print(chars.takeIndexInterval(2)); // [1, 3, 5, 7, 9]
+  /// print(chars.takeIndexInterval(2, offset: 1)); // [2, 4, 6, 8]
+  /// ```
+  List<E> indexIntervals(int n, {int offset = 0}) {
```

Comment:
What do you think about matching the naming of `Ranges` and rename it to `step`?

```dart
for (var i in 10.rangeTo(2).step(2)) {
  print(i); // 10, 8, 6, 4, 2
}
```

I'd remove the `offset` because the same thing could be achieved with `skip(n)`

Reply:
I like step a lot better. As you can see in the discrepancy between the comment and the method name, I was having some trouble naming the method.

I'll remove offset.